server/src/main/java/org/elasticsearch/node/Node.java

```diff
@@ -437,6 +441,14 @@ protected Node(
                                                  namedWriteableRegistry).stream())
                 .collect(Collectors.toList());
 
+            NodeIdPatternConverter nodeIdPatternConverter = NodeIdPatternConverter.newInstance(new String[]{});
```

Comment:
very often I see ClusteStateListeners adding themselves to ClusterService with clusterService.addListener(this) inside a constructor. I don't think this a good idea as leaking the this reference during construction can cause race conditions and visibility problems.
I guess we should add them once constructed like here in 445